clap_generate/src/generators/mod.rs

```diff
@@ -44,24 +44,24 @@ pub trait Generator {
     /// pub struct ClapDebug;
     ///
     /// impl Generator for ClapDebug {
-    ///     fn generate(app: &App, buf: &mut dyn Write) {
+    ///     fn generate(&self, app: &App, buf: &mut dyn Write) {
     ///         write!(buf, "{}", app).unwrap();
     ///     }
-    /// #   fn file_name(name: &str) -> String {
+    /// #   fn file_name(&self, name: &str) -> String {
     /// #    name.into()
     /// #   }
     /// }
     /// ```
-    fn generate(app: &App, buf: &mut dyn Write);
+    fn generate(&self, app: &App, buf: &mut dyn Write);
 
     /// Gets all subcommands including child subcommands in the form of `("name", "bin_name")`.
     ///
     /// Subcommand `rustup toolchain install` would be converted to
     /// `("install", "rustup toolchain install")`.
-    fn all_subcommands(app: &App) -> Vec<(String, String)> {
-        let mut subcmds: Vec<_> = Self::subcommands(app);
+    fn all_subcommands(&self, app: &App) -> Vec<(String, String)> {
```

Comment:
While I do agree that `file_name` and `generate` should take `&self`, all these functions that deal with purely `App` should be static since they will never depend on `&self`.

Actually, I don't understand why they were in `Generator` trait in the first place.

Reply:
I agree about moving them out, but I see that best being left for another PR.

Reply:
> Actually, I don't understand why they were in Generator trait in the first place.

I'm not sure exactly. I don't believe I created the Generator trait, so its hard for me to say. I imagine it was originally because if you're generating something from `clap` you're like 99% generating something *about* an `App` struct.

Reply:
Ah, I missed that question. I believe that is also independent of this PR and shouldn't block it.

With that said, it seems like having some kind of trait is helpful for simplifying composing and dispatching to interoperable generators.   Only taking a `fn` is limited (nicer to take a trait that has a blanket impl for `fnt`) and this has two functions, so I'd be curious what your thought is for a non-trait version.  Maybe post that as a Discussion or Issue?